admin/services/check-window.service.js

```diff
@@ -142,6 +142,39 @@ const checkWindowService = {
     })
 
     return R.concat(newCheckWindows, pastCheckWindows)
+  },
+
+  /**
+   * Get editable check window
+   * @param {Number} id
+   * @returns {Object}
+   */
+  getEditableCheckWindow: async(id) => {
+    if (!id) {
+      throw new Error('Check window id not provided')
+    }
+    const checkWindow = await checkWindowDataService.sqlFindOneById(id)
+    const currentDate = moment().format('YYYY-MM-D')
+    const adminStartDate = moment(checkWindow.adminStartDate, 'D MM YYYY').format('YYYY-MM-D')
+    const checkStartDate = moment(checkWindow.checkStartDate, 'D MM YYYY').format('YYYY-MM-D')
+
+    return {
+      checkWindowId: id,
+      checkWindowName: checkWindow.name,
+      adminStartDay: moment(checkWindow.adminStartDate).format('D'),
```

Comment:
Why are the dates re-parsed every time?  Wouldn't it make sense here to parse them once with moment and _then_  extract the date-parts with format?

Reply:
@jon-shipley updated